File: Sep12/Lecture7Examples/Example25/MainActivity.java
package com.varunshankar.example25;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.net.URL;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        String location = "Salt&Lake&City,us";
        URL weatherDataURL = NetworkUtils.buildURLFromString(location);
        String jsonWeatherData = null;
        try{
            jsonWeatherData = NetworkUtils.getDataFromURL(weatherDataURL);
        }catch(Exception e){
            e.printStackTrace();
        }

    }


}

File: Sep12/Example25/app/src/main/java/com/varunshankar/example25/MainActivity.kt
package com.varunshankar.example25

import com.varunshankar.example25.NetworkUtils.buildURLFromString
import com.varunshankar.example25.NetworkUtils.getDataFromURL
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import java.lang.Exception

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val location = "Salt&Lake&City,us"
        val weatherDataURL = buildURLFromString(location)
        var jsonWeatherData: String? = null
        try {
            assert(weatherDataURL != null)
            jsonWeatherData = getDataFromURL(weatherDataURL!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}